magpie/setup/magpie-setup-project-hive

```diff
@@ -0,0 +1,203 @@
+#############################################################################
+#  Copyright (C) 2013-2015 Lawrence Livermore National Security, LLC.
+#  Produced at Lawrence Livermore National Laboratory (cf, DISCLAIMER).
+#  Written by Albert Chu <chu11@llnl.gov>
+#  LLNL-CODE-644248
+#  
+#  This file is part of Magpie, scripts for running Hadoop on
+#  traditional HPC systems.  For details, see https://github.com/llnl/magpie.
+#  
+#  Magpie is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#  
+#  Magpie is distributed in the hope that it will be useful, but
+#  WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#  
+#  You should have received a copy of the GNU General Public License
+#  along with Magpie.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+
+# This script sets up configuration files for jobs.  For the most
+# part, it shouldn't be editted.  See job submission files for
+# configuration details.
+
+if [ "${HIVE_SETUP}" != "yes" ]
+then
+    exit 0
+fi
+
+source ${MAGPIE_SCRIPTS_HOME}/magpie/exports/magpie-exports-dirs
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-node-identification
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-setup
+
+if ! Magpie_am_I_a_hive_node
+then
+   exit 0
+fi
+Magpie_calculate_hadoop_filesystem_paths ${hivenoderank}
+
+if [ "${HIVE_LOCAL_SCRATCH_DIR}X" == "X" ]
+then
+    hivelocalscratchdirpath="${magpie_hadooptmpdir}/hive"
+
+    if [ ! -d "${hivelocalscratchdirpath}" ]
+    then
+        mkdir -p ${hivelocalscratchdirpath}
+        if [ $? -ne 0 ] ; then
+            echo "mkdir failed making ${hivelocalscratchdirpath}"
+            exit 1
+        fi
+    fi
+else
+    IFSORIG=${IFS}
+    IFS=","
+    hivelocalscratchdirpath=""
+    for hivelocalpath in ${HIVE_LOCAL_SCRATCH_DIR}
+    do
+        # Setup node directory per node in case not a local drive
+        hivelocalpathtmp="${hivelocalpath}/node-${hivenoderank}"
+
+        if [ ! -d "${hivelocalpathtmp}" ]
+        then
+            mkdir -p ${hivelocalpathtmp}
+            if [ $? -ne 0 ] ; then
+                echo "mkdir failed making ${hivelocalpathtmp}"
+                exit 1
+            fi
+        fi
+
+        hivelocalscratchdirpath="${hivelocalscratchdirpath}${hivelocalscratchdirpath:+","}${hivelocalpathtmp}"
+    done
+    IFS=${IFSORIG}
+fi
+
+if [ "${HIVE_PORT}X" != "X" ]
+then
+    hiveport=${HIVE_PORT}
+else
+    hiveport=10000
+fi
+
+if [ "${HIVE_METASTORE_DIR}X" != "X" ]
+then
+    hivemetastoredir=${HIVE_METASTORE_DIR}
+else
+    hivemetastoredir="${hivelocalscratchdirpath}/metastore"
+fi
+
+if [ "${PGDATA}X" != "X" ]
```

Comment:
tabbing here and next few if branches is off.

Reply:
Fixed, thanks.